include/ginkgo/core/base/executor.hpp

```diff
@@ -1108,14 +1165,17 @@ class CudaExecutor : public detail::ExecutorBase<CudaExecutor>,
         return num_execs[device_id];
     }
 
+    void populate_exec_info(
+        std::shared_ptr<const MachineTopology> mach_topo) override;
```

Comment:
```suggestion
        std::shared_ptr<const MachineTopology>& mach_topo) override;
```
Or only use the pointer.